language/move-vm/types/src/views.rs

```diff
@@ -17,11 +20,131 @@ pub trait TypeView {
 /// This is used to expose certain info to clients (e.g. the gas meter),
 /// usually in a lazily evaluated fashion.
 pub trait ValueView {
+    fn visit(&self, visitor: &mut impl ValueVisitor);
+
     /// Returns the abstract memory size of the value.
     ///
     /// The concept of abstract memory size is not well-defined and is only kept for backward compatibility.
     /// New applications should avoid using this.
-    fn legacy_abstract_memory_size(&self) -> AbstractMemorySize;
+    ///
+    /// TODO(Gas): Encourage clients to replicate this in their own repo and get this removed once
+    ///            they are done.
+    fn legacy_abstract_memory_size(&self) -> AbstractMemorySize {
```

Comment:
@jolestar I would recommend you replicate this in the Starcoin repo. Better, you might also want to revise the formula a bit since some of the numbers here picked for no obvious reasons. It might also be a good idea to store some of the parameters on-chain so you can adjust them dynamically.

Reply:
cc @LemonHX

Reply:
cc @oxade